src/components/DeckMechanisms/index.js

```diff
@@ -89,21 +89,21 @@ export default class DeckMechanisms extends React.Component {
       // in the hand, and that are not the one that has been cycled. From there,
       // we can draw a random card while taking weight into account, then push
       // the new card into the hand.
-      const availableCards = state.deck
-        .filter(card => !state.hand.includes(card.id))
-        .filter(card => card.id !== id)
+      const availableCards = state.deck.filter(
+        card => !state.hand.includes(card.id)
+      )
       const pick = rwc(availableCards)
       newState.hand.push(pick)
 
       // After having drawn a new card, we need to readjust the weight of all
       // cards that are not in the hand, as well as the one that has just been
-      // drawn (reseted to 0).
+      // drawn (reset to 0).
       newState.deck = this.getIncreasedDeckWeight({
         state: newState,
         reset: [id, pick],
       })
 
-      newState.hasCycledThisTurn = id !== 'N22' // Ignore Goldgrubbers cycling
```

Comment:
This previously caused the issue of **playing** Goldgrubbers making them count as **cycled**, since id refers to the card that's being cycled. Another issue appeared when playing Snake Eyes, as this would set `newState.hasCycledThisTurn` to true.
Adding a second argument solves both issues